ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -277,6 +298,21 @@ public Integer getValue() {
   }
 
   /**
+   * Initializes a {@link Gauge} metric for the maxPutChunkSize. This value is a constant for a running router
+   * instance, but can change between invocations. This metric helps maintain historical information.
+   * @param maxPutChunkSize the value to initialize this metric with.
+   */
+  public void initializeMaxPutChunkSizeMetric(final int maxPutChunkSize) {
+    maxPutChunkSizeGauge = new Gauge<Integer>() {
+      @Override
+      public Integer getValue() {
+        return maxPutChunkSize;
+      }
+    };
+    metricRegistry.register(MetricRegistry.name(NonBlockingRouter.class, "MaxPutChunkSizeBytes"), maxPutChunkSizeGauge);
```

Comment:
If we play around with chunk sizes, won't the commit history have the details we need? 
It is not harmful because of the computation cost (which is minimal) but because of the idea that a configuration has to be exported as a metric to keep track of history of a certain config - you can present the same point for _any_ configuration value.

Also keep in mind that the metric itself is ephemeral - sure you can have another product that maintains the history of the metric (just like commit histories) but the metric itself is not useful for the purpose you mention.